pulp_container/app/viewsets.py

```diff
@@ -594,16 +594,17 @@ def add(self, request, pk):
         """
         Queues a task that creates a new RepositoryVersion by adding content units.
         """
-        add_content_units = []
         repository = self.get_object()
         serializer = serializers.RecursiveManageSerializer(data=request.data)
         serializer.is_valid(raise_exception=True)
 
         if "content_units" in request.data:
-            for url in request.data["content_units"]:
-                content = NamedModelViewSet.get_resource(url, Content)
-                content.touch()
-                add_content_units.append(str(content.pk))
+            add_content_units = [
```

Comment:
@lubosmj  tldr; you'd need to make more or less same changes as you did in core for the `modify` endpoint. Where you've also raise error for non existent content in the separate function you've written.